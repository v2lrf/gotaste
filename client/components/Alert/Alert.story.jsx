import React, { Fragment } from 'react'
import { storiesOf } from '@storybook/react'

import Alert from '.'

storiesOf('Alert', module)
  .add('Info', () => (
    <div className="p-16">
      <Alert kind="info">Hey I just want you to know that I love wine!</Alert>
    </div>
  ))
  .add('Danger', () => (
    <div className="p-16">
      <Alert kind="danger">
        Whoa! We are out of wine. Hurry up and go get some
      </Alert>
    </div>
  ))
  .add('Floating', () => (
    <Fragment>
      <Alert kind="info" fixed>
        Hey! Look at me, I am floating
      </Alert>
      <br />
      <br />
      <br />
      <br />
      <br />
      <br />
      <br />
      <br />
      <br />
      <br />
      <br />
      <br />
      <br />
      <br />
      <br />
      <br />
      <br />
      <br />
      <br />
      <br />
      <br />
      <br />
      <br />
      <br />
      <br />
      <br />
      <br />
      <br />
      <br />
      <br />
      <br />
      <br />
      <br />
      <br />
      <br />
      <br />
      <br />
      <br />
      <br />
      <br />
      <br />
      <br />
      <br />
      <br />
      <br />
      <br />
      <br />
      <br />
      <br />
      <br />
      <br />
      <br />
      <br />
      <br />
      <br />
      <br />
      <br />
      <br />
      <br />
      <br />
      <br />
      <br />
      <br />
      <br />
      <br />
      <br />
      <br />
      <br />
      <br />
      <br />
      <br />
      <br />
      <br />
      <br />
    </Fragment>
  ))
