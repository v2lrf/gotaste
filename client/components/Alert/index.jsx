import React from 'react'
import PropTypes from 'prop-types'
import classNames from 'classnames'
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome'
import {
  faExclamationCircle,
  faInfoCircle
} from '@fortawesome/pro-light-svg-icons'

function Alert({ kind, children }) {
  const alertClasses = classNames(
    'flex items-center border shadow rounded my-4 py-2 text-sm',
    {
      'bg-red-lightest text-red-darkest border-red-light': kind === 'danger',
      'bg-white text-red-darkest border-red-lighter': kind === 'info'
    }
  )
  return (
    <div className={alertClasses}>
      <span className="pl-2 mr-2" role="img" aria-label="ikon" aria-hidden>
        {kind === 'info' && (
          <FontAwesomeIcon
            icon={faInfoCircle}
            className="text-red-light text-xl"
          />
        )}
        {kind === 'danger' && (
          <FontAwesomeIcon
            icon={faExclamationCircle}
            className="text-red-light text-xl"
          />
        )}
      </span>
      <div className="pr-2">{children}</div>
    </div>
  )
}

Alert.propTypes = {
  kind: PropTypes.oneOf(['info', 'danger']).isRequired,
  children: PropTypes.node.isRequired
}

export default Alert
