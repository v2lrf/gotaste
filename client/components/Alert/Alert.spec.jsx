import React from 'react'
import Alert from '.'

describe('Alert', () => {
  let component
  let props
  beforeEach(() => {
    props = {
      kind: 'info'
    }

    component = shallow(<Alert {...props}>Whoa! I am an alert.</Alert>)
  })

  describe('Snapshots', () => {
    xit('should match the info snapshot', () => {
      expect(component).toMatchSnapshot()
    })

    xit('should match the danger snapshot', () => {
      component.setProps({ kind: 'danger' })
      expect(component).toMatchSnapshot()
    })

    xit('should match the fixed snapshot', () => {
      component.setProps({ fixed: true })
      expect(component).toMatchSnapshot()
    })
  })
})
