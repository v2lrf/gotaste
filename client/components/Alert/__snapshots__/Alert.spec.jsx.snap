// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Alert Snapshots should match the danger snapshot 1`] = `
<div
  className="flex items-center border shadow rounded my-4 py-2 text-sm bg-red-lightest text-red-darkest border-red-light"
>
  <span
    aria-hidden={true}
    aria-label="ikon"
    className="pl-2 mr-2"
    role="img"
  >
    <FontAwesomeIcon
      border={false}
      className="text-red-light text-xl"
      fixedWidth={false}
      flip={null}
      icon={
        Object {
          "icon": Array [
            512,
            512,
            Array [],
            "f06a",
            "M256 40c118.621 0 216 96.075 216 216 0 119.291-96.61 216-216 216-119.244 0-216-96.562-216-216 0-119.203 96.602-216 216-216m0-32C119.043 8 8 119.083 8 256c0 136.997 111.043 248 248 248s248-111.003 248-248C504 119.083 392.957 8 256 8zm-11.49 120h22.979c6.823 0 12.274 5.682 11.99 12.5l-7 168c-.268 6.428-5.556 11.5-11.99 11.5h-8.979c-6.433 0-11.722-5.073-11.99-11.5l-7-168c-.283-6.818 5.167-12.5 11.99-12.5zM256 340c-15.464 0-28 12.536-28 28s12.536 28 28 28 28-12.536 28-28-12.536-28-28-28z",
          ],
          "iconName": "exclamation-circle",
          "prefix": "fal",
        }
      }
      inverse={false}
      listItem={false}
      mask={null}
      pull={null}
      pulse={false}
      rotation={null}
      size={null}
      spin={false}
      symbol={false}
      title=""
      transform={null}
    />
  </span>
  <div
    className="pr-2"
  >
    Whoa! I am an alert.
  </div>
</div>
`;

exports[`Alert Snapshots should match the info snapshot 1`] = `
<div
  className="flex items-center border shadow rounded my-4 py-2 text-sm bg-white text-red-darkest border-red-lighter"
>
  <span
    aria-hidden={true}
    aria-label="ikon"
    className="pl-2 mr-2"
    role="img"
  >
    <FontAwesomeIcon
      border={false}
      className="text-red-light text-xl"
      fixedWidth={false}
      flip={null}
      icon={
        Object {
          "icon": Array [
            512,
            512,
            Array [],
            "f05a",
            "M256 40c118.621 0 216 96.075 216 216 0 119.291-96.61 216-216 216-119.244 0-216-96.562-216-216 0-119.203 96.602-216 216-216m0-32C119.043 8 8 119.083 8 256c0 136.997 111.043 248 248 248s248-111.003 248-248C504 119.083 392.957 8 256 8zm-36 344h12V232h-12c-6.627 0-12-5.373-12-12v-8c0-6.627 5.373-12 12-12h48c6.627 0 12 5.373 12 12v140h12c6.627 0 12 5.373 12 12v8c0 6.627-5.373 12-12 12h-72c-6.627 0-12-5.373-12-12v-8c0-6.627 5.373-12 12-12zm36-240c-17.673 0-32 14.327-32 32s14.327 32 32 32 32-14.327 32-32-14.327-32-32-32z",
          ],
          "iconName": "info-circle",
          "prefix": "fal",
        }
      }
      inverse={false}
      listItem={false}
      mask={null}
      pull={null}
      pulse={false}
      rotation={null}
      size={null}
      spin={false}
      symbol={false}
      title=""
      transform={null}
    />
  </span>
  <div
    className="pr-2"
  >
    Whoa! I am an alert.
  </div>
</div>
`;
